include/openrave/kinbody.h

```diff
@@ -3066,6 +3083,25 @@ class OPENRAVE_API KinBody : public InterfaceBase
     ConfigurationSpecification _spec;
     CollisionCheckerBasePtr _selfcollisionchecker; ///< optional checker to use for self-collisions
 
+public:
+    // Gary registers forward kinematics functions
+    struct ForwardKinematicsStruct {
+        ForwardKinematicsStruct();
+        ModuleBaseConstPtr pCalculatorModule = nullptr; ///< kinbody basic calculators module
+        boost::function<bool(const std::vector<double>&)> pSetLinkTransformsFn; ///< function that sets links' transforms
+        boost::function<void(std::vector<double>&)> pGetDOFLastSetValuesFn; ///< function that updates kinbody's dof values
+        bool bInitialized = false; ///< indicator of successful initialization
```

Comment:
@tgn3000 what is the use of this? why not just check if `pSetLinkTransformsFn` is valid?

Reply:
`bInitialized==true` means all FK functions are properly initialized

Reply:
@tgn3000 isn't that the same as checking `!!pSetLinkTransformsFn && !!pGetDOFLastSetValuesFn`?

Reply:
ok